cpp/src/decisiontree/batched-levelalgo/kernels.cuh

```diff
@@ -240,7 +240,9 @@ __global__ void nodeSplitKernel(IdxT max_depth, IdxT min_rows_per_node,
   auto range_start = node->start, range_len = node->count;
   auto isLeaf = leafBasedOnParams<DataT, IdxT>(
     node->depth, max_depth, min_rows_per_node, max_leaves, n_leaves, range_len);
-  if (isLeaf || splits[nid].best_metric_val <= min_impurity_decrease) {
+  if (isLeaf || splits[nid].best_metric_val <= min_impurity_decrease
+      || splits[nid].nLeft < min_rows_per_node
```

Comment:
minor thing: typically `nvcc` is smart enough not to issue 2 load requests for this and the one below this line. But it's safer (and explicit) if you first load this into a variable and use it instead in the conditional here.

Reply:
Good point.